TestTracer.test_start_span_multi_intertwined fails randomly

```
_________________ TestTracer.test_start_span_multi_intertwined _________________

self = <tests.opentracer.test_tracer.TestTracer object at 0x7fbad9a33240>
ot_tracer = <ddtrace.opentracer.tracer.Tracer object at 0x7fbad9a3bcc0>
writer = <tests.utils.tracer.DummyWriter object at 0x7fbad9a3b358>

    def test_start_span_multi_intertwined(self, ot_tracer, writer):
        """Start multiple spans at the top level intertwined.
        Alternate calling between two traces.
        """
        import threading
        import time
    
        def trace_one():
            id = 11
            with ot_tracer.start_active_span(str(id)):
                id += 1
                time.sleep(0.009)
                with ot_tracer.start_active_span(str(id)):
                    id += 1
                    time.sleep(0.001)
                    with ot_tracer.start_active_span(str(id)):
                        pass
    
        def trace_two():
            id = 21
            with ot_tracer.start_active_span(str(id)):
                id += 1
                time.sleep(0.006)
                with ot_tracer.start_active_span(str(id)):
                    id += 1
                    time.sleep(0.009)
                with ot_tracer.start_active_span(str(id)):
                    pass
    
        # the ordering should be
        # t1.span1/t2.span1, t2.span2, t1.span2, t1.span3, t2.span3
        t1 = threading.Thread(target=trace_one)
        t1.daemon = True
        t2 = threading.Thread(target=trace_two)
        t2.daemon = True
    
        t1.start()
        t2.start()
        # wait for threads to finish
        time.sleep(0.018)
    
        spans = writer.pop()
    
        # trace_one will finish before trace_two so its spans should be written
        # before the spans from trace_two, let's confirm this
>       assert spans[0].name == '11'
E       AssertionError: assert '21' == '11'
E         - 21
E         + 11
```

--- a/tests/opentracer/test_tracer.py
+++ b/tests/opentracer/test_tracer.py
@@ -296,23 +296,35 @@ class TestTracer(object):
         """
         import threading
 
+        # synchronize threads with a threading event object
+        event = threading.Event()
+
         def trace_one():
             id = 11
             with ot_tracer.start_active_span(str(id)):
+                event.set()
                 id += 1
+                event.wait()
                 with ot_tracer.start_active_span(str(id)):
+                    event.set()
                     id += 1
+                    event.wait()
                     with ot_tracer.start_active_span(str(id)):
-                        pass
+                        event.set()
 
         def trace_two():
             id = 21
+            event.wait()
             with ot_tracer.start_active_span(str(id)):
+                event.set()
                 id += 1
+                event.wait()
                 with ot_tracer.start_active_span(str(id)):
+                    event.set()
                     id += 1
+                event.wait()
                 with ot_tracer.start_active_span(str(id)):
-                    pass
+                    event.set()
 
         # the ordering should be
         # t1.span1/t2.span1, t2.span2, t1.span2, t1.span3, t2.span3

